lib/schemas.js

```diff
@@ -147,8 +147,8 @@ function getSchemaAnyway (schema, jsonShorthand) {
  *
  * @param {object} context the request context
  * @param {number} statusCode the http status code
- * @param {string} contentType the reply content type
- * @returns {function|boolean} the right JSON Schema function to serialize
+ * @param {string} [contentType] the reply content type
+ * @returns {function|false} the right JSON Schema function to serialize
```

Comment:
`@returns {function|boolean}` is correct. The expected value should be documented in the text.

Reply:
{function|false} is actuallY correct, as we dont return true but the truthy value of function. This resulted in the place were we call this function that typescript jsdoc typing system threw a typing error, because it said that the result is potentially true and that is not a function.

Reply:
The proposed change is not valid jsdoc. The return "type" should be an actual type, not an instance of one.

Reply:
I reference the spec defined at https://jsdoc.app/

Reply:
I would say that they are different dialects of the same thing, and that one has not had a substantial update in almost 6 years: https://github.com/jsdoc/jsdoc.github.io/commits/master

Where as the TS dialect of it has: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

It all depends on where one aims for the JSDoc to be used – if its TS/VSCode, then follow that dialect, if its eg. a documentation generator, then follow whatever dialect it supports.